sql/analyzer/optimization_rules.go

```diff
@@ -83,123 +53,65 @@ func moveJoinConditionsToFilter(ctx *sql.Context, a *Analyzer, n sql.Node, scope
 		return n, transform.SameTree, nil
 	}
 
-	var nonJoinFilters []sql.Expression
-	var topJoin sql.Node
-	node, same, err := transform.Node(n, func(n sql.Node) (sql.Node, transform.TreeIdentity, error) {
+	return transform.Node(n, func(n sql.Node) (sql.Node, transform.TreeIdentity, error) {
```

Comment:
Seems like this rule might be better off as a function during join planning, rather than a discrete rule, makes it much harder to express it independently like this

Could be a good followup step